elide-core/src/main/java/com/yahoo/elide/core/PersistentResource.java

```diff
@@ -894,22 +899,19 @@ public String getType() {
 
     @Override
     public int hashCode() {
-        final int prime = 31;
-        int result = 1;
-        String id = dictionary.getId(getObject());
-        result = prime * result + (uuid.isPresent() ? uuid.hashCode() : 0);
-        result = prime * result + (id == null ? 0 : id.hashCode());
-        result = prime * result + (type == null ? 0 : type.hashCode());
-        return result;
+        if (getObject() == null) {
+            return -1;
+        }
+        return getObject().getClass().hashCode();
```

Comment:
Perhaps this should be `getObject().getType().hashCode()` since object _may_ be a proxy